src/DecoratorManifold.jl

```diff
@@ -221,29 +224,30 @@ macro decorator_transparent_function(fallback_case, input_ex)
                 $(callargs[2:end]...);
                 $(kwargs_list...),
             ) where {$(where_exprs...)}
-                return ($fname)(
-                    $(argnames[1]),
-                    ManifoldsBase._acts_transparently($fname, $(argnames...)),
-                    $(argnames[2:end]...),
-                    ;
-                    $(kwargs_call...),
-                )
+                transparency = ManifoldsBase._acts_transparently($fname, $(argnames...))
+                if transparency === Val(:parent)
```

Comment:
Can this maybe be done with a dispatch? Maybe the function is not the nicest (it also need `parts`, `argnames` and `kwargs_call`), but it would be nicer and a little more Julia-style-ish?

Reply:
Hmm... maybe? I'll try turning that `if` into dispatch.

Reply:
As far as I can see it would require a lot of methods like
```julia
get_dispatch_function(::typeof(exp!), ::Val{:transparent}) = exp!__transparent
```
One method for each function and transparency type. Do you think it's worth it?

Reply:
I had hoped it would suffice to do one function per mode, like
````
get_dispatch_function(f, ::Val{:transparent}, parts, kwarg-names) = ...
````
where this is before you quote it again so you can build up the function name and pass it back? But I also did not try, that's why I carefully asked.

Reply:
So, you mean dispatch at macroexpansion? That wouldn't give us anything since the macro wouldn't be re-run each time `get_dispatch_function` gets new methods.

Reply:
You're right. Then we can also stick with the ifs, though currently that's doubled, right?

Reply:
Yes, it's duplicated but there is enough differences around the duplicated block that removing this duplication would almost certainly add complexity.

Reply:
Ok, then it's fine. I only saw the block added in a quite similar manner and did not yet check their surrounding code.